cyral/resource_cyral_repository_user_account.go

```diff
@@ -0,0 +1,596 @@
+			map[string]interface{}{
+				"kubernetes_secret": []interface{}{
+					map[string]interface{}{
+						"secret_name": resource.AuthScheme.KubernetesSecret.SecretName,
+						"secret_key":  resource.AuthScheme.KubernetesSecret.SecretKey,
+					},
+				},
+			},
+		}
+	} else {
+		return fmt.Errorf("auth scheme is required, user account is corrupt: %v", resource)
+	}
+
+	if err := d.Set("auth_scheme", authScheme); err != nil {
+		return fmt.Errorf("error setting 'auth_scheme': %w", err)
+	}
+
+	return nil
+}
+
+// ReadFromSchema is used to translate a .tf file into whatever the
+// UserAccounts API expects.
+func (userAccount *UserAccountResource) ReadFromSchema(d *schema.ResourceData) error {
+	// Set basic values required fields (UserID is computed).
+	userAccount.Name = d.Get("name").(string)
+	userAccount.AuthDatabaseName = d.Get("auth_database_name").(string)
+
+	// Handle approval config (optional field).
+	approvalConfig := d.Get("approval_config").(*schema.Set).List()
+	if len(approvalConfig) > 0 {
+		m := approvalConfig[0].(map[string]interface{})
+		userAccount.Config = &UserAccountConfig{
+			Approval: &ApprovalConfig{},
+		}
+		autogrant, ok := m["automatic_grant"]
+		if ok {
+			userAccount.Config.Approval.AutomaticGrant =
+				autogrant.(bool)
+		}
+		maxAutoGrant, ok := m["max_auto_grant_duration"]
+		if ok {
+			userAccount.Config.Approval.MaxAutomaticGrantDuration =
+				maxAutoGrant.(string)
+		}
+	}
+
+	// Handle Auth Scheme (required field).
+	authSchemeSet := d.Get("auth_scheme").(*schema.Set).List()
+	if len(authSchemeSet) != 1 {
+		return fmt.Errorf(
+			"exactly one auth_scheme attribute is required",
+		)
+	}
+
+	authSchemes := authSchemeSet[0].(map[string]interface{})
+
+	for k, v := range authSchemes {
+		authSchemeDetails := v.(*schema.Set).List()
+		if len(authSchemeDetails) == 0 {
+			continue
+		}
+		if len(authSchemeDetails) > 1 {
+			return fmt.Errorf(
+				"exactly one of %v is required in auth_scheme",
+				allAuthSchemes,
+			)
+		}
+		m := authSchemeDetails[0].(map[string]interface{})
+
+		switch k {
+		case "environment_variable":
+			userAccount.AuthScheme = &AuthScheme{
+				EnvironmentVariable: &AuthSchemeEnvironmentVariable{
+					VariableName: m["variable_name"].(string),
+				},
+			}
+		case "aws_iam":
+			userAccount.AuthScheme = &AuthScheme{
+				AWSIAM: &AuthSchemeAWSIAM{
+					RoleARN: m["role_arn"].(string),
+				},
+			}
+		case "aws_secrets_manager":
+			userAccount.AuthScheme = &AuthScheme{
+				AWSSecretsManager: &AuthSchemeAWSSecretsManager{
+					SecretArn: m["secret_arn"].(string),
+				},
+			}
+		case "cyral_storage":
+			userAccount.AuthScheme = &AuthScheme{
+				CyralStorage: &AuthSchemeCyralStorage{
+					Password: m["password"].(string),
+				},
+			}
+		case "hashicorp_vault":
+			userAccount.AuthScheme = &AuthScheme{
+				HashicorpVault: &AuthSchemeHashicorpVault{
+					Path: m["path"].(string),
+				},
+			}
+			if m["is_dynamic_user_account"] != nil {
```

Comment:
`is_dynamic_user_account` should probably be a required field.

Reply:
...and this `if` should be removed. :)